dot/parachain/availability-distribution/session_cache.go

```diff
@@ -150,6 +163,37 @@ func (c *LRUSessionCache) GetSessionInfo(
 	return cachedSessionInfo, nil
 }
 
+func (c *LRUSessionCache) GetAuthorityID(
+	validatorIndex parachaintypes.ValidatorIndex,
+	relayParent common.Hash,
+	rt runtime.Instance,
+) (parachaintypes.AuthorityDiscoveryID, error) {
+	authIDs := c.authIDCache.Get(relayParent)
+
+	if authIDs == nil {
+		sessionIndex, err := rt.ParachainHostSessionIndexForChild()
+		if err != nil {
+			return parachaintypes.AuthorityDiscoveryID{}, err
+		}
+
+		sessionInfo, err := rt.ParachainHostSessionInfo(sessionIndex)
+		if err != nil {
+			return parachaintypes.AuthorityDiscoveryID{}, err
+		}
+
+		authIDs = sessionInfo.DiscoveryKeys
+	}
+
+	c.authIDCache.Put(relayParent, authIDs)
```

Comment:
should not the put call be inside the `if authIDs == nil`? otherwise we will always put no matter if the info is already cached.

Reply:
Yes it should!